### PR TITLE
Add plugin system [ECR-3622, ECR-3623, ECR-3624]

### PR DESCRIPTION
**Pull request by popzxc**
tl;dr: with this PR merged you can add support of custom runtimes and create custom config loaders for instances.

Full description:
This PR introduces a plugin system with the following concrete features:

- Declaring new runtimes by calling `Configuration.declare_runtime(runtime_name, runtime_id)`.

- Runtime-specific artifact spec loaders. You can implement spec loader for your runtime by inheriting `RuntimeSpecLoader` base class, and then provide it to `Launcher` with `add_runtime_spec_loader` method. Launcher then will use this loader to encode runtime-specific artifact spec into payload.

- Artifact-specific instance config loaders. Same as artifact spec loader, but you can create loader specifig to concrete service. Then Launcher will use provided loader to encode instance config into bytes.

Also this PR adds a default `InstanceSpecLoader` (instead of `get_service_config` method) which will attempt to recursively parse the `Config` message.

### CODE REVIEW

File: exonum_launcher/instances/default_spec_loader.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    def load_spec(self, loader: ProtobufLoader, instance: Instance) -> bytes:
+        try:
+            try:
+                # Try to load module (if it's already compiled) first.
+                service_module = ModuleManager.import_service_module(instance.artifact.name, "service")
+            except (ModuleNotFoundError, ImportError):
+                # If it's not compiled, load & compile protobuf.
+                loader.load_service_proto_files(instance.artifact.runtime_id, instance.artifact.name)
+                service_module = ModuleManager.import_service_module(instance.artifact.name, "service")
```
Comment (aleksuss) [inline exonum_launcher/instances/default_spec_loader.py:26]: It looks a little bit weird as for me. What if add some checker into the `Loader` for checking corresponding proto files ? And finally get something like this:

```python
if not loader.is_exist_service_proto_file(instance.artifact.name):
    loader.load_service_proto_files(instance.artifact.runtime_id, instance.artifact.name)

service_module = ModuleManager.import_service_module(instance.artifact.name, "service")
```

Comment (popzxc) [reply exonum_launcher/instances/default_spec_loader.py:26]: Calling the code and test if it raises an exception is a common way tro check things in python, [for example](https://stackoverflow.com/a/319298).

However, it definetely makes sense to add method like `is_loaded` to `ProtobufLoader` and then rewrite this part.

I'll do it in the next PR to the light client.

**Review by aleksuss — APPROVED** [review]

**Review by alekseysidorov — APPROVED** [review]
